client/app/services/query.js

```diff
@@ -43,6 +43,40 @@ class QueryResultError {
 }
 
 
+class Parameter {
+  constructor(parameter) {
+    this.title = parameter.title;
+    this.name = parameter.name;
+    this.type = parameter.type;
+    this.value = parameter.value;
+    this.global = parameter.global;
+  }
+
+  get ngModel() {
+    if (this.type === 'date' || this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
+      this.$$value = this.$$value || moment(this.value).toDate();
+      return this.$$value;
+    } else if (this.type === 'number') {
+      this.$$value = this.$$value || parseInt(this.value, 10);
+      return this.$$value;
+    }
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
```

Comment:
For now this maintains backward compatibility with v0.12.0 and fixes a regression. We can revisit this and add option to set the format in a future pull request.

But I don't think it should be in an env variable, but rather as configuration for the parameter, as users might use different databases on the same Redash deployment.